### ISSUE TITLE
Drop support for Ubuntu 14.04 / 16.04 / CentOS 7 on Bazel@HEAD

### ISSUE THREAD

**Issue by davido** [bazelbuild/bazel#13561]
Considering that Ubuntu 16.04 LTS is EOL now, it might be reasonable to drop support for Ubuntu 14.04 / 16.04 / CentOS 7 on Bazel's main branch.

One reason to discontinue support for outdated platforms, is that Bazel@HEAD cannot be built on modern platforms, like Fedora 34 rawhide (x86_64) that switched to `gcc` 11.x toolchain (https://gcc.gnu.org/gcc-11/), see https://github.com/bazelbuild/bazel/issues/12702.

The attempt to bump transitive dependencies in Bazel for `gRPC` and `absl` caused the build to fail on outdated CentOS 7 platform due to outdated 4.8.5 C++ compiler toolchain:

```
  $ docker run -it gcr.io/bazel-public/centos7-java8:latest /bin/bash
  $ gcc --version
  gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
```

See CI breakages for this PR: https://github.com/bazelbuild/bazel/pull/13536.

**Comment by philwo** [comment]
@davido Thanks! I transferred the issue to the Bazel project.

**Comment by philwo** [comment]
I sent a mail to bazel-discuss@ to get feedback on this: https://groups.google.com/g/bazel-discuss/c/xXkCsFjosG8

**Comment by Mizux** [comment]
TLDR: With Centos-7 you can have a recent gcc (9) with an "old" glibcxx while ubuntu pull the libstdc++ 3.4.28

You can have a gcc 9 on Centos 7 using this:
https://github.com/google/or-tools/blob/b37d9c786b69128f3505f15beca09e89bf078a89/tools/docker/centos-7.Dockerfile#L12-L20 and still have a good old glibcxx

It can be a good idea to keep centos-7 since it is the image used by manylinux to build native python package (e.g. Google OR-Tools with our SWIG python wrapper).
note: centos-6 use a now forbidden system call so docker image won't work on any recent distro/glinux etc...

And unfortunately Ubuntu ppa toolchain-r pull 3.4.28 -_-

| distro | GCC |GLIBC | GLIBCXX |
| -- | -- | -- | -- |
| Ubuntu 16.04 LTS | 5.4.0 |2.23 | 3.4.21 |
| Ubuntu 16.04 LTS + gcc-7 | 7.5.0 | 2.23 | 3.4.28 |
| Ubuntu 16.04 LTS + gcc-9 | 9.3.0 | 2.23 | 3.4.28 |
| Ubuntu 18.04 LTS | 7.5.0 | 2.27 | 3.4.25 |
| Centos 7 | 4.8.5 | 2.17 | 3.4.19 |
| **Centos 7 + gcc-9** | 9.3.1 | 2.17 | 3.4.19 |
| Centos 8 | 8.3.1 | 2.28 | 3.4.25 |
| Debian 9 | 6.3.0 | 2.24 | 3.4.22 |
| Debian 10 | 8.3.0 | 2.28 | 3.4.25 |

note: [ubuntu-toolchain-r](https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/test?field.series_filter=xenial) seems to pull libstdc++ 28 -_-

note:
`gcc --version`
`find / -type f -iname "libstdc++*.so*"`
`find / -type f -iname "libc*.so*"`
`objdump -p libXXX.so`

ref: https://github.com/google/or-tools/issues/2520#issuecomment-825119882

**Comment by davido** [comment]
@Mizux

Thanks for your comments. Especially the pointer to CentOS 7 with gcc 9 toolchain.

I have created a minimal CentOS 7 with gcc 9 toolchain Dockerfile: [1], created docker image from it and verified, that I can built Bazel with my [PR](https://github.com/bazelbuild/bazel/pull/13536) included, where `gPRC` and `Abseil` were bumped to recent releases.

[1] https://github.com/davido/centos7-gcc-9-bazel-docker/blob/master/Dockerfile

**Comment by meteorcloudy** [comment]
This is already fixed. We have dropped vanilla centos 7, the minimal gcc version is 10.2.1 and we no longer test on ubuntu 14, and 16.
